rpc/src/v1/tests/mocked/personal.rs

```diff
@@ -96,29 +96,40 @@ fn new_account() {
 	assert_eq!(res, Some(response));
 }
 
-#[test]
-fn sign_and_send_transaction_with_invalid_password() {
+fn invalid_password_test(method: &str)
+{
 	let tester = setup();
 	let address = tester.accounts.new_account("password123").unwrap();
+
 	let request = r#"{
 		"jsonrpc": "2.0",
-		"method": "personal_sendTransaction",
+		"method": ""#.to_owned() + method + r#"",
 		"params": [{
-			"from": ""#.to_owned() + format!("0x{:?}", address).as_ref() + r#"",
+			"from": ""# + format!("0x{:?}", address).as_ref() + r#"",
 			"to": "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
 			"gas": "0x76c0",
 			"gasPrice": "0x9184e72a000",
 			"value": "0x9184e72a"
 		}, "password321"],
 		"id": 1
-	}"#;
+	}"#;	
 
 	let response = r#"{"jsonrpc":"2.0","error":{"code":-32021,"message":"Account password is invalid or account does not exist.","data":"SStore(InvalidPassword)"},"id":1}"#;
 
 	assert_eq!(tester.io.handle_request_sync(request.as_ref()), Some(response.into()));
 }
 
 #[test]
+fn sign_transaction_with_invalid_password() {
```

Comment:
Would be nice to have a positive test case as well.

Reply:
Don't sure how to check if the response is actually valid.

Is there a convenient way to validate the whole JSON response, knowing that it should describe a transaction? I.e. all fields should exist and types should be correct? Maybe I could try to deserialize it?

Reply:
Yeah, deserializing it makes sense.